From 58939441a597793a34ca7e16dc966095e44e35e0 Mon Sep 17 00:00:00 2001
From: Jiri Pirko <jpirko@redhat.com>
Date: Mon, 14 Jul 2014 11:47:57 -0400
Subject: [net] gre6: don't try to add the same route two times

Message-id: <1405338479-17101-1-git-send-email-jpirko@redhat.com>
Patchwork-id: 85736
O-Subject: [patch rhel7 net 1/3] sit/gre6: don't try to add the same route two times
Bugzilla: 1107369
RH-Acked-by: David S. Miller <davem@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Hannes Frederic Sowa <hannes@redhat.com>

BZ1107369
https://brewweb.devel.redhat.com/taskinfo?taskID=7696184

upstream commit f7cb8886335dea39fa31bb701700361f1aa7a6ea
Author: Nicolas Dichtel <nicolas.dichtel@6wind.com>
Date:   Thu Nov 14 13:51:07 2013 +0100

    sit/gre6: don't try to add the same route two times

    addrconf_add_linklocal() already adds the link local route, so there is no
    reason to add it before calling this function.

Signed-off-by: Jiri Pirko <jpirko@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/ipv6/addrconf.c b/net/ipv6/addrconf.c
index cf25455..1b61479 100644
--- a/net/ipv6/addrconf.c
+++ b/net/ipv6/addrconf.c
@@ -2731,7 +2731,6 @@ static void addrconf_sit_config(struct net_device *dev)
   struct in6_addr addr;
 
   ipv6_addr_set(&addr,  htonl(0xFE800000), 0, 0, 0);
-  addrconf_prefix_route(&addr, 64, dev, 0, 0);
   if (!ipv6_generate_eui64(addr.s6_addr + 8, dev))
    addrconf_add_linklocal(idev, &addr);
   return;
@@ -2762,8 +2761,6 @@ static void addrconf_gre_config(struct net_device *dev)
  }
 
  ipv6_addr_set(&addr,  htonl(0xFE800000), 0, 0, 0);
- addrconf_prefix_route(&addr, 64, dev, 0, 0);
-
  if (!ipv6_generate_eui64(addr.s6_addr + 8, dev))
   addrconf_add_linklocal(idev, &addr);
 }
-- 
1.7.1